sway-lib-core/src/never.sw

```diff
@@ -2,6 +2,25 @@ library never;
 
 use ::ops::{Not, Eq, Ord};
 
+/// `Never` is assigned to a block that is deterministically aborted. This can be
+/// achieved by using `return`, `break`, `continue` or `revert` inside a block.
+///
+/// An example of such block is:
+///
+/// ```
+/// fn foo() -> u32 {
+///     let b: u32 = {
+///         return 5;
+///     };
+///     123
+/// }
+/// ```
+///
+/// In the example above the assigning of `return 5` to `b` is pointless because
+/// `return 5` returns from the entire function. Nonetheless internally the return
+/// block will be considered of type `Never`. Furthermore this example shows also
+/// that `Never` type can be coerced into any other type, here `b` is of type `u32`
```

Comment:
"this example shows also that `Never` type can be coerced into any other type". This bit is pretty important, can you add docstrings to the `impl`'s for `Never` and include this as a note?

Reply:
Looked into Sway and Rust documentation and usually the docstring are only attached to traits not to impls.
Added note about `Never` type coercion.